lib/pure/math.nim

```diff
@@ -163,6 +163,24 @@ proc prod*[T](x: openArray[T]): T {.noSideEffect.} =
   result = 1.T
   for i in items(x): result = result * i
 
+proc cumsum*[T](x: openArray[T]): seq[T] =
+  ## Return cumulative aka prefix summation of ``x``.
+  ##
+  ## .. code-block:: nim
+  ##   var x = [1, 2, 3, 4]
+  ##   echo x.cumsum    # [1, 3, 6, 10]
+  result.setLen(x.len)
+  result[0] = x[0]
+  for i in 1 ..< x.len: result[i] = result[i-1] + x[i]
+
+proc cumsum*[T](x: var openArray[T]) =
```

Comment:
This overload doesn't work well, ``var x = [1, 2]; let y = cumsum(x)`` --> no overloading by return value in Nim.

Reply:
Ok.  I pushed a fix with tests for both variants instead of just testing one.  I also changed the name to `cumsummed` (since "summed" with two 'm's is the right tense there for "sum").  Assuming I didn't mess up any tests we should be good to go.